src/processor/map/map_copy_from.cpp

```diff
@@ -90,100 +78,43 @@ std::unique_ptr<PhysicalOperator> PlanMapper::mapCopyNodeFrom(LogicalOperator* l
-    auto directedInMemRelData = std::make_unique<DirectedInMemRelData>();
-    auto boundTableID = schema->getBoundTableID(direction);
-    auto numNodes =
-        nodesStore.getNodesStatisticsAndDeletedIDs()->getMaxNodeOffsetPerTable().at(boundTableID) +
-        1;
-    if (schema->isSingleMultiplicityInDirection(direction)) {
-        // columns.
-        auto relColumns = std::make_unique<DirectedInMemRelColumns>();
-        relColumns->adjColumn =
-            std::make_unique<InMemColumn>(StorageUtils::getAdjColumnFName(outputDirectory,
-                                              schema->tableID, direction, DBFileType::ORIGINAL),
-                LogicalType(LogicalTypeID::INTERNAL_ID));
-        relColumns->adjColumnChunk =
-            relColumns->adjColumn->createInMemColumnChunk(0, numNodes - 1, csvReaderConfig);
-        for (auto i = 0u; i < schema->getNumProperties(); ++i) {
-            auto propertyID = schema->properties[i]->getPropertyID();
-            auto propertyDataType = schema->properties[i]->getDataType();
-            auto fName = StorageUtils::getRelPropertyColumnFName(
-                outputDirectory, schema->tableID, direction, propertyID, DBFileType::ORIGINAL);
-            relColumns->propertyColumns.emplace(
-                propertyID, std::make_unique<InMemColumn>(fName, *propertyDataType));
-            relColumns->propertyColumnChunks.emplace(
-                propertyID, relColumns->propertyColumns.at(propertyID)
-                                ->createInMemColumnChunk(0, numNodes - 1, csvReaderConfig));
-        }
-        directedInMemRelData->setColumns(std::move(relColumns));
-    } else {
-        // lists.
-        auto relLists = std::make_unique<DirectedInMemRelLists>();
-        relLists->adjList =
-            std::make_unique<InMemAdjLists>(StorageUtils::getAdjListsFName(outputDirectory,
-                                                schema->tableID, direction, DBFileType::ORIGINAL),
-                numNodes);
-        relLists->relListsSizes = std::make_unique<atomic_uint64_vec_t>(numNodes);
-        for (auto i = 0u; i < schema->getNumProperties(); ++i) {
-            auto property = schema->getProperty(i);
-            auto fName = StorageUtils::getRelPropertyListsFName(outputDirectory, schema->tableID,
-                direction, property->getPropertyID(), DBFileType::ORIGINAL);
-            relLists->propertyLists.emplace(property->getPropertyID(),
-                InMemListsFactory::getInMemPropertyLists(fName, *property->getDataType(), numNodes,
-                    csvReaderConfig, relLists->adjList->getListHeadersBuilder()));
-        }
-        directedInMemRelData->setRelLists(std::move(relLists));
-    }
-    return directedInMemRelData;
+    auto dataColumnPositions = getExpressionsDataPos(copyFromInfo->dataColumnsToCopy, *outFSchema);
+    auto dataFormat = tableSchema->isSingleMultiplicityInDirection(direction) ?
```

Comment:
open an issue about removing `isSingleMultiplicityInDirection` for me

Reply:
See #2321